src/plugins/bookmark/bookmark.js

```diff
@@ -18,48 +22,51 @@
  * Copyright 2016-2017 Henrik Ingo (@henrikingo)
  * Released under the MIT license.
  */
-/* global window, document, impress */
+/* global document, impress */
 
-( function( document, window ) {
+( function( document ) {
     "use strict";
-    var api;
     var hotkeys = {};
-    function hotkeyDest(event) {
-	return (hotkeys.hasOwnProperty(event.key)  ? hotkeys[event.key] :
-		hotkeys.hasOwnProperty(event.code) ? hotkeys[event.code] : null) }
-    
+    function hotkeyDest( event ) {
+	return ( hotkeys.hasOwnProperty( event.key )  ? hotkeys[ event.key ] :
+		 hotkeys.hasOwnProperty( event.code ) ? hotkeys[ event.code ] : null ); }
+
     // In pre-init phase, build a map of bookmark hotkey to div id, by reviewing all steps
     impress.addPreInitPlugin( function( root, api ) {
-	root.querySelectorAll( ".step" ).forEach(function(div) {
-            if ( div.dataset.bookmarkKeyList !== undefined && div.id != undefined) {
-		div.dataset.bookmarkKeyList.split( " " ).forEach( (k) => {
-		    if (hotkeys.hasOwnProperty(k)) hotkeys[k].push( div.id ); // if hotkeys collide, we'll cycle through the targets
-		    else hotkeys[k] = [div.id]; } ); } });
+	root.querySelectorAll( ".step" ).forEach( function( div ) {
+            if ( div.dataset.bookmarkKeyList !== undefined && div.id !== undefined ) {
+		div.dataset.bookmarkKeyList.split( " " ).forEach( ( k ) => {
+		    if ( hotkeys.hasOwnProperty( k ) ) {
+			hotkeys[ k ].push( div.id );
+		    } else { hotkeys[ k ] = [ div.id ]; } } ); } } );
 
 	api.lib.gc.addEventListener( document, "keyup", function( event ) {
-	    if (hotkeyDest(event) != undefined) {
+	    if ( hotkeyDest( event ) !== undefined ) {
 		event.stopImmediatePropagation();
-		api.next(event);
-		// event.preventDefault();
+		api.next( event );
+
+		// Event.preventDefault();
 	    }
-	});
-    });
+	} );
+    } );
 
     // In pre-stepleave phase, match a hotkey and reset destination accordingly.
     impress.addPreStepLeavePlugin( function( event ) {
-	// window.console.log(`bookmark: running as PreStepLeavePlugin; event=`); window.console.log(event)
+
+	// Window.console.log(`bookmark: running as PreStepLeavePlugin; event=`);
+	// window.console.log(event)
         if ( ( !event || !event.origEvent ) ) { return; }
-	var dest = hotkeyDest(event.origEvent)
-	// window.console.log(`bookmark: recognizing hotkey ${event.origEvent.code} goes to ${dest}`)
+	var dest = hotkeyDest( event.origEvent );
         if ( dest ) {
-	    // window.console.log(`bookmark: recognizing hotkey ${event.code} goes to ${dest}`)
-            var newTarget = document.getElementById( dest[0] ); // jshint ignore:line
+
+	    // Window.console.log(`bookmark: recognizing hotkey ${event.code} goes to ${dest}`)
```

Comment:
Do you really want to leave the debug code commented out in here?

Reply:
i have taken it out, will push a new rev. Still working on the test failure.